homeassistant/components/logbook/__init__.py

```diff
@@ -414,6 +416,19 @@ def yield_events(query):
                     & (States.state != old_state.state)
                 )
             )
+            #
+            # Prefilter out continuous domains that have
+            # ATTR_UNIT_OF_MEASUREMENT as its much faster in sql.
+            #
+            .filter(
+                (Events.event_type != EVENT_STATE_CHANGED)
+                | (
+                    sqlalchemy.not_(States.domain.in_(CONTINUOUS_DOMAINS))
+                    | sqlalchemy.not_(
+                        States.attributes.contains(UNIT_OF_MEASUREMENT_JSON)
+                    )
```

Comment:
I think that we can remove the extra nesting:

- event_type != state_changed OR
- state not in continuous_domains OR
- uom_json in attributes

Reply:
Yes, that works.  Good call 👍